pkg/platform/runtime/artifact/download.go

```diff
@@ -45,13 +45,17 @@ func NewDownloadsFromCamelBuild(buildStatus *headchef_models.V1BuildStatusRespon
 			if strings.HasSuffix(a.URI.String(), ".tar.gz") || strings.HasSuffix(a.URI.String(), ".zip") {
 				return []ArtifactDownload{{ArtifactID: *a.ArtifactID, UnsignedURI: a.URI.String(), UnsignedLogURI: a.LogURI.String(), Checksum: a.Checksum}}, nil
 			}
-
 		}
 	}
 
-	if buildStatus.Type != nil && *buildStatus.Type == headchef_models.V1BuildStatusResponseTypeBuildStarted {
+	if buildStatusType := buildStatus.Type; buildStatusType != nil {
 		logging.Debug("buildStatus=%v", buildStatus)
-		return nil, CamelRuntimeBuilding
+		switch {
+		case *buildStatusType == headchef_models.V1BuildStatusResponseTypeBuildStarted:
+			return nil, CamelRuntimeBuilding
+		case *buildStatusType == headchef_models.V1BuildStatusResponseTypeBuildFailed:
+			return nil, errs.New("Build error: %+v", buildStatus)
```

Comment:
I think this should be localized, the error might be helpful for the user.

Reply:
I went ahead and localized this message, but there is a note about errors from headchef: "Note that these errors may not be suitable for presenting to users and should simply be logged for further investigation."